File: AngularCoinShare/src/app/services/transaction.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Guid } from '../models';
import { PaginationDto, TransactionDto, PersonOverviewDto } from '../dtos';
import { map } from 'rxjs/operators';

@Injectable({
    providedIn: 'root'
})
export class TransactionService {
    private readonly apiBaseUrl: string = "https://localhost:5001/api/transaction";

    constructor(private http: HttpClient) { }

    public getByGroupId(groupId: Guid, paginationConfig: { offset: number, take: number }): Observable<PaginationDto<TransactionDto[]>> {
        const { offset, take } = paginationConfig;
        return this.http.get<PaginationDto<TransactionDto[]>>(`${this.apiBaseUrl}/group/${groupId}?offset=${offset}&take=${take}`).pipe(
            map(t => {
                t.items.forEach(i => i.paidOn = new Date(i.paidOn));
                return t;
            })
        );
    }

    public getPersonOverviews(groupId: Guid): Observable<PersonOverviewDto[]> {
        return this.http.get<PersonOverviewDto[]>(`${this.apiBaseUrl}/group/${groupId}/persons`);
    }

    public create(transaction: TransactionDto): Observable<number> {
        return this.http.post<number>(this.apiBaseUrl, transaction);
    }
}

File: BlazorCoinShare/CoinShare.Web/Services/TransactionService.cs
using System;
using System.Collections.Generic;
using System.Net.Http;
using System.Threading.Tasks;
using CoinShare.Api.Shared.Dtos;
using Microsoft.AspNetCore.Components;

namespace CoinShare.Web.Services
{
    public class TransactionService
    {
        private readonly string apiBaseUrl = "https://localhost:5001/api/transaction";
        private readonly HttpClient httpClient;

        public TransactionService(HttpClient httpClient)
        {
            this.httpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
        }

        public Task<PaginationDto<TransactionDto>> GetByGroupId(Guid groupId, int offset, int take)
        {
            return this.httpClient
                .GetJsonAsync<PaginationDto<TransactionDto>>($"{apiBaseUrl}/group/{groupId}?offset={offset}&take={take}");
        }

        public Task<IList<PersonOverviewDto>> GetPersonOverviewsAsync(Guid groupId)
        {
            return this.httpClient.GetJsonAsync<IList<PersonOverviewDto>>($"{apiBaseUrl}/group/{groupId}/persons");
        }

        public Task<int> CreateAsync(TransactionDto transaction)
        {
            return this.httpClient.PostJsonAsync<int>(apiBaseUrl, transaction);
        }
    }
}
